src/Post.php

```diff
@@ -513,8 +513,10 @@ public function getKeywordsStrAttribute()
      */
     public function newFromBuilder($attributes = [], $connection = null)
     {
-        if (array_key_exists($attributes->post_type, static::$postTypes)) {
+        if (is_object($attributes) && array_key_exists($attributes->post_type, static::$postTypes)) {
             $class = static::$postTypes[$attributes->post_type];
+        } elseif (is_array($attributes) && array_key_exists($attributes['post_type'], static::$postTypes)) {
+            $class = static::$postTypes[$attributes['post_type']];
```

Comment:
You could simplify this by casting the array as an object...
```php
        if ($attributes && array_key_exists($attributes['post_type'], static::$postTypes)) {
            $class = static::$postTypes[((object) $attributes)->post_type];
        }
```

For example...
```php
<?php

class Post {
    public $post_type;
  
    public function __construct($post_type)
    {
        $this->post_type = $post_type;
    }
}

$str = 'hello world!';
$post_type = 'post';
$test = ['post' => $str];

$arr = ['post_type' => $post_type];
$obj = (object) ['post_type' => $post_type];
$post = new Post($post_type);

var_dump($test[((object) $arr)->post_type]); // string(12) "hello world!"
var_dump($test[((object) $obj)->post_type]); // string(12) "hello world!"
var_dump($test[((object) $post)->post_type]); // string(12) "hello world!"
```